### PR TITLE
#167 GUM Managing Tool

### PR DESCRIPTION
**Pull request by r-mishev**

### CODE REVIEW

File: cli.sh [hunk 1]
```diff
@@ -49,12 +49,6 @@ elif [ "$ACTIONS" == "$DEPLOY" ]; then
...
 
     if [ "$ACTIONS" == "$LOGIN_IN_VM" ]; then
-    echo "What's the user of the $(gum style --foreground 212 "Virtual Machine") (in most cases it's $(gum style --foreground 212 "root")):"
-    read USER 
-
-    echo "What's the ip of the $(gum style --foreground 212 "Virtual Machine")":
-    read IP 
-
-    ssh "$USER@$IP";
+        ssh $HUB_VM
```
Comment (asynchroza) [inline cli.sh:52]: The idea for this is to check if $HUB_VM has a value and do `ssh $HUB_VM` when there is one, otherwise old steps are to be executed.

You need a second option which is going to replace DEPLOY_SPA which will essentially going to do the same with the difference that it's going to run the `set_vm_env` script when it successfully connects to the VM

Comment (r-mishev) [reply cli.sh:52]: Should I bother doing the second option now or do I leave it for later? I think it'd be best for me to do it prior to us actually deploying the website tomorrow

File: cli.sh [hunk 2]
```diff
@@ -49,12 +49,29 @@ elif [ "$ACTIONS" == "$DEPLOY" ]; then
...
+    
+    elif [ "$ACTIONS" == "$DEPLOY_SPA" ]; then
+        if [[ -z "${HUB_VM}" ]]; then
+            echo "What's the user of the $(gum style --foreground 212 "Virtual Machine") (in most cases it's $(gum style --foreground 212 "root")):"
+            read USER
+
+            echo "What's the IP of the $(gum style --foreground 212 "Virtual Machine"):"
+            read IP
+
+            ssh "$USER@$IP" "curl https://raw.githubusercontent.com/AUBGTheHUB/spa-website-2022/master/set_vm_env.sh | bash"
```
Comment (asynchroza) [inline cli.sh:72]: lacks privileges

File: README.md [hunk 3]
```diff
@@ -62,6 +62,36 @@ cd ~/go/src/spa-website-2022 && make post-wsl
...
+    * ``Run Client (requests towards deployed API)``
+    * ``Run API``
+    * ``SSH into a VM``
+    * ``Set up VM for Deployment``
+
+*If you don't want to use gum, you may proceed to the next sections*
+#### __Set up__:
+```
+1. vim ~/.bashrc
+2. Add export HUB_VM="root@188.166.65.120" (make sure it's not in an if statement or for cycle)
```
Comment (asynchroza) [inline README.md:77]: Remember what we discussed to for the IPs? Don't put them here.

File: cli.sh [hunk 4]
```diff
@@ -44,17 +44,34 @@ if [ $ACTIONS == $START ]; then
...
+    
+    elif [ "$ACTIONS" == "$SET_VM_ENV" ]; then
+        if [[ -z "${HUB_VM}" ]]; then
+            echo "What's the user of the $(gum style --foreground 212 "Virtual Machine") (in most cases it's $(gum style --foreground 212 "root")):"
+            read USER
+
+            echo "What's the IP of the $(gum style --foreground 212 "Virtual Machine"):"
+            read IP
+
+            ssh "$USER@$IP" "sudo curl https://raw.githubusercontent.com/AUBGTheHUB/spa-website-2022/master/set_vm_env.sh | bash"
```
Comment (asynchroza) [inline cli.sh:72]: If you verify if this is working?

File: cli.sh [hunk 5]
```diff
@@ -44,17 +44,34 @@ if [ $ACTIONS == $START ]; then
 
 elif [ "$ACTIONS" == "$DEPLOY" ]; then
     LOGIN_IN_VM="SSH into a Virtual Machine"
-    DEPLOY_SPA="Deploy SPA on a Virtual Machine"
+    SET_VM_ENV="Set up Virtual Machine for Deployment"
 
-    ACTIONS=$(gum choose --cursor-prefix "[ ] " --selected-prefix "[✓] " --no-limit "$LOGIN_IN_VM" "$DEPLOY_SPA")
+    ACTIONS=$(gum choose --cursor-prefix "[ ] " --no-limit "$LOGIN_IN_VM" "$SET_VM_ENV")
```
Comment (asynchroza) [inline cli.sh:49]: Why did you remove the selected prefix?

File: README.md (hunk 3 above)
Comment (r-mishev) [reply README.md:77]: I thought of this, but how to I tell them what to put as a variable?

File: cli.sh (hunk 5 above)
Comment (r-mishev) [reply cli.sh:49]: I supposed I had to remove it. Did you mean I should've just changed the icon (a.k.a the tick mark)?

File: cli.sh (hunk 4 above)
Comment (r-mishev) [reply cli.sh:72]: It does work

File: cli.sh [hunk 6]
```diff
@@ -49,12 +49,29 @@ elif [ "$ACTIONS" == "$DEPLOY" ]; then
     ACTIONS=$(gum choose --cursor-prefix "[ ] " --selected-prefix "[✓] " --no-limit "$LOGIN_IN_VM" "$DEPLOY_SPA")
```
[the remark this answers is not in the document]
Comment (asynchroza) [reply cli.sh]: change the icon in the brackets
